### ISSUE TITLE
Add httpstan information to compile function

### ISSUE THREAD

**Issue by ahartikainen** [stan-dev/httpstan#446]
Add library information to tempfile in compile function.

In https://github.com/stan-dev/httpstan/blob/main/httpstan/compile.py

```
with tempfile.NamedTemporaryFile() as fh:
```

Should we add information for suffix (`.stan`) and prefix (`httpstan`), so debugging messages could have a bit more informative messages.

e.g. the created c++ code has

```
static const std::vector<string> locations_array__ = {" (found before start of program)",
                                                      " (in '/tmp/tmp30gvcel0', line 2, column 2 to column 9)",
                                                      " (in '/tmp/tmp30gvcel0', line 5, column 2 to column 18)"};
```

**Comment by riddell-stan** [comment]
That's a good idea.

We might also consider naming the file {stan_model_name}.stan in the 
temporary directory.

Maybe fix https://github.com/stan-dev/httpstan/issues/425 in the same 
commit?

**Comment by riddell-stan** [comment]
Closed in #447
